packages/graph-explorer/src/hooks/useDeleteConfig.ts

```diff
@@ -6,52 +6,52 @@ import {
   schemaAtom,
 } from "@/core";
 import { logger } from "@/utils";
+import { useAtomValue } from "jotai";
+import { useAtomCallback } from "jotai/utils";
 import { useCallback } from "react";
-import { useRecoilCallback, useRecoilValue } from "recoil";
 
 export function useDeleteConfig() {
-  return useRecoilCallback(
-    ({ set }) =>
-      (id: ConfigurationId) => {
-        logger.log("Deleting connection:", id);
-        set(activeConfigurationAtom, prev => {
-          if (prev === id) {
-            return null;
-          }
-          return prev;
-        });
+  return useAtomCallback(
+    useCallback(async (_get, set, id: ConfigurationId) => {
+      logger.log("Deleting connection:", id);
+      await set(activeConfigurationAtom, async prev => {
```

Comment:
When setting storage atoms it is very important to make sure everything is awaited. This caused me some headaches, because the compiler does not enforce awaiting the call to `set()`.

Reply:
I did a search and found some places where I am not awaiting the `set`. These are likely not causing a runtime issue, but I'm going to go ahead and fix them anyway.

Reply:
I enabled the ESLint rule that checks async await calls and it found hundreds of issues. I'll dedicate an entire PR after this one to resolve them all. For now, let's focus on getting this PR merged.